internal/middleware/compression_test.go

```diff
@@ -79,10 +118,14 @@ func TestHTTPClientCompression(t *testing.T) {
 			require.NoError(t, err, "failed to create request to test handler")
 
 			client := http.Client{}
-			if tt.encoding == "gzip" {
-				client.Transport = NewCompressRoundTripper(http.DefaultTransport)
+			if tt.encoding != "" {
+				client.Transport = NewCompressRoundTripper(http.DefaultTransport, tt.encoding)
 			}
 			res, err := client.Do(req)
```

Comment:
Continue to the comment in NewBodyReader(), in the case of `gzip` and `zlib`, this `client.Do(req)` function returns `code 400` with no error, while `snappy` and `zstd` returns error with no `res`, which is unexpected.